operator/internal/controller/redpanda/nodepool_controller.go

```diff
@@ -92,5 +111,58 @@ func (r *NodePoolReconciler) Reconcile(ctx context.Context, req ctrl.Request) (r
 		return ctrl.Result{Requeue: true}, nil
 	}
 
+	var status statuses.NodePoolStatus
+	sts, err := statefulSetForNodePool(ctx, r.Client, pool)
```

Comment:
This is kinda funky. Just to make sure I understand it correct, we're using an index here because we want to re-trigger reconcilation for node pools when the STS changes. No such index is required for the Redpanda resource because that's what's actually performing the reconcilation?